src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -332,88 +333,133 @@ export class BrowsingContextProcessor {
+          // OOPiF.
+          maybeContext.updateCdpTarget(cdpTarget);
+        } else {
+          // New context.
+          BrowsingContextImpl.create(
+            cdpTarget,
+            this.#realmStorage,
+            targetInfo.targetId,
+            null,
+            this.#eventManager,
+            this.#browsingContextStorage,
+            this.#logger
+          );
+        }
+        return;
+      }
+      case 'worker': {
+        this.#setEventListeners(targetCdpClient);
+
+        const cdpTarget = CdpTarget.create(
+          targetInfo.targetId,
+          targetCdpClient,
+          this.#browserCdpClient,
+          sessionId,
+          this.#realmStorage,
+          this.#eventManager,
+          this.#preloadScriptStorage,
+          this.#networkStorage
+        );
 
-    const cdpTarget = CdpTarget.create(
-      targetInfo.targetId,
-      targetCdpClient,
-      this.#browserCdpClient,
-      sessionId,
-      this.#realmStorage,
-      this.#eventManager,
-      this.#preloadScriptStorage,
-      this.#networkStorage
-    );
+        this.#createWorkerRealm(cdpTarget);
+        return;
+      }
+    }
 
-    if (maybeContext) {
-      // OOPiF.
-      maybeContext.updateCdpTarget(cdpTarget);
-    } else {
-      // New context.
-      BrowsingContextImpl.create(
-        cdpTarget,
+    // DevTools or some other not supported by BiDi target. Just release
+    // debugger and ignore them.
+    targetCdpClient
+      .sendCommand('Runtime.runIfWaitingForDebugger')
+      .then(() =>
+        parentSessionCdpClient.sendCommand('Target.detachFromTarget', params)
+      )
+      .catch((error) => this.#logger?.(LogType.debugError, error));
+  }
+
+  #workers = new Map<string, Realm>();
+  #createWorkerRealm(cdpTarget: CdpTarget) {
+    cdpTarget.cdpClient.on('Runtime.executionContextCreated', (params) => {
+      const {uniqueId, id, origin} = params.context;
+      const realm = new Realm(
         this.#realmStorage,
-        targetInfo.targetId,
-        null,
-        this.#eventManager,
         this.#browsingContextStorage,
+        uniqueId,
+        cdpTarget.targetId,
+        id,
+        serializeOrigin(origin),
+        'dedicated-worker',
+        undefined,
+        cdpTarget.cdpClient,
+        this.#eventManager,
         this.#logger
       );
-    }
+      this.#workers.set(cdpTarget.cdpSessionId, realm);
+    });
   }
 
   #handleDetachedFromTargetEvent(
     params: Protocol.Target.DetachedFromTargetEvent
   ) {
-    // XXX: params.targetId is deprecated. Update this class to track using
-    // params.sessionId instead.
-    // https://github.com/GoogleChromeLabs/chromium-bidi/issues/60
-    const contextId = params.targetId!;
-    this.#browsingContextStorage.findContext(contextId)?.dispose();
-
-    this.#preloadScriptStorage
-      .find({targetId: contextId})
-      .map((preloadScript) => preloadScript.dispose(contextId));
+    const context = this.#browsingContextStorage
```

Comment:
dispose is not called anymore on the context?

Reply:
It is called? Next line.